packages/snaps-cli/src/commands/eval/implementation.test.ts

```diff
@@ -0,0 +1,78 @@
+import * as utils from '@metamask/snaps-utils';
+import { SnapEvalError } from '@metamask/snaps-utils';
+import { DEFAULT_SNAP_BUNDLE } from '@metamask/snaps-utils/test-utils';
+import normalFs from 'fs';
+import { dirname, join } from 'path';
+
+import { getCompiler } from '../../webpack';
+import { evaluate } from './implementation';
+
+const { promises: fs } = normalFs;
+
+jest.mock('fs');
+
+jest.mock('@metamask/snaps-utils', () => ({
+  ...jest.requireActual('@metamask/snaps-utils'),
+  evalBundle: jest
+    .fn()
+    .mockImplementation(jest.requireActual('@metamask/snaps-utils').evalBundle),
+}));
+
+jest.mock('../../webpack', () => ({
+  ...jest.requireActual('../../webpack'),
+  getCompiler: jest.fn<
+    ReturnType<typeof getCompiler>,
+    Parameters<typeof getCompiler>
+  >((...args) => {
+    const compiler = jest
+      .requireActual<typeof import('../../webpack')>('../../webpack')
+      .getCompiler(...args);
+
+    compiler.inputFileSystem = normalFs;
+    compiler.outputFileSystem = normalFs;
+
+    return compiler;
+  }),
+}));
+
+describe('evaluate', () => {
+  const BUNDLE_PATH = join(__dirname, '__test__', 'webpack', 'good', 'eval.js');
+
+  beforeAll(async () => {
+    // `snaps-utils` uses the fake `fs` module, so we need to create the
+    // directory and file.
+    await fs.mkdir(dirname(BUNDLE_PATH), { recursive: true });
+    await fs.writeFile(BUNDLE_PATH, DEFAULT_SNAP_BUNDLE);
+  });
+
+  it('evaluates the bundle', async () => {
+    // Since the eval runs in a separate process, we can't mock the file system
+    // and need to use an actual file.
+    const output = await evaluate(BUNDLE_PATH);
+
+    expect(output.stdout).toBe('');
+    expect(output.stderr).toBe('');
+  });
+
+  it('rejects if the bundle has an error', async () => {
+    jest.spyOn(utils, 'evalBundle').mockImplementation(() => {
+      throw new SnapEvalError('Eval error.', {
```

Comment:
Couldn't we just use `webpack/bad/eval.js` instead of mocking?

Reply:
This is running in a mocked file system, so mocking the function call seems easier.